src/foundry/client/pixi/placeables/drawing.d.ts

```diff
@@ -1,204 +1,184 @@
-     * http://scaledinnovation.com/analytics/splines/aboutSplines.html
-     * @param factor   - The smoothing factor
-     * @param previous - The prior point
-     * @param point    - The current point
-     * @param next     - The next point
-     * @internal
+     * Prepare the text style used to instantiate a PIXI.Text or PreciseText instance for this Drawing document.
      */
-    protected _getBezierControlPoints(
-      factor: number,
-      previous: [number, number],
-      point: [number, number],
-      next: [number, number],
-    ): {
-      cp1: {
-        x: number;
-        y: number;
-      };
-      next_cp0: {
-        x: number;
-        y: number;
-      };
-    };
+    protected _getTextStyle(): PIXI.TextStyle;
 
-    /**
-     * Refresh the boundary frame which outlines the Drawing shape
-     * @internal
-     */
-    protected _refreshFrame({ x, y, width, height }: Rectangle): void;
+    protected override _applyRenderFlags(flags: Drawing.RenderFlags): void;
 
     /**
      * Add a new polygon point to the drawing, ensuring it differs from the last one
-     * @param temporary - (default: `true`)
+     * @param position - The drawing point to add
+     * @param options  - Options which configure how the point is added
      * @internal
      */
-    protected _addPoint(position: Point, temporary?: boolean): void;
+    protected _addPoint(
+      position: Point,
+      options?: {
+        /**
+         * Should the point be rounded to integer coordinates?
+         * @defaultValue `false`
+         */
+        round?: boolean;
+
+        /**
+         * Should the point be snapped to grid precision?
+         * @defaultValue `false`
+         */
+        snap?: boolean;
+
+        /**
+         * Is this a temporary control point?
+         * @defaultValue `false`
+         */
+        temporary?: boolean;
+      },
+    ): void;
 
     /**
      * Remove the last fixed point from the polygon
      * @internal
      */
     protected _removePoint(): void;
 
-    protected override _onControl(options: PlaceableObject.ControlOptions & { isNew?: boolean }): void;
+    protected override _onControl(options: PlaceableObject.ControlOptions & Drawing.TextEditingOptions): void;
 
     protected override _onRelease(options: PlaceableObject.ReleaseOptions): void;
 
     protected override _onDelete(options: DocumentModificationOptions, userId: string): void;
 
-    /**
-     * Handle text entry in an active text tool
-     * @internal
-     */
-    protected _onDrawingTextKeydown(event: KeyboardEvent): void;
-
+    // TODO: Replace after data model
     protected override _onUpdate(data: DeepPartial<foundry.data.DrawingData["_source"]>): void;
 
+    override activateListeners(): void;
+
     /**
      * @param event - unused
      */
+    // TODO: Replace User reference after data model
     protected override _canControl(user: InstanceType<ConfiguredDocumentClassForName<"User">>, event?: any): boolean;
```

Comment:
I think this should work mostly the same way post data model changes, it's going to be an instance of `User` basically right?

Reply:
Yeah it's a `User` document. So remove the TODO?

Reply:
Yup.